### PR TITLE
Not updating registration_sent value when updating user.

### LINKED ISSUE #5445: PUT /users/:id sets tm_user.registration_sent (date) to null
## I'm submitting a ...
-  bug report

## Traffic Control components affected ...
-  Traffic Ops

## Current behavior:
PUT /users/:id sets tm_user.registration_sent (date) to null

## Expected behavior:
PUT /users/:id should NOT touch tm_user.registration_sent (date) in any way. if it has a value, leave it. if it has no value, leave it. 

tm_user.registration_sent is set if a user is created via `POST users/register` and should never be nulled once it is set. it can, however, be updated if `POST users/register` is called again with the same email BUT it should never be nulled.

## Minimal reproduction of the problem with instructions:
- Create a user via `POST users/register` and provide email, tenant and role
- check the database for the newly created user. see that registration_sent is set to current date/time
- note the id of the newly created user
- update the newly created user via `PUT /user/:id` and notice that registration_sent is set to null. this is wrong.

### PR DESCRIPTION
**Pull request by rimashah25**
## What does this PR (Pull Request) do?

- [x] This PR fixes #5445  

## Which Traffic Control components are affected by this PR?

- Traffic Ops

## What is the best way to verify this PR?

Run API tests

## If this is a bug fix, what versions of Traffic Control are affected?

master (9930731)

## The following criteria are ALL met by this PR

- [x] I have explained why tests are unnecessary
- [x] I have explained why documentation is unnecessary
- [x] This PR includes an update to CHANGELOG.md OR such an update is not necessary
- [x] This PR includes any and all required license headers
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY** (see [the Apache Software Foundation's security guidelines](https://www.apache.org/security/) for details)

## Additional Information

No change in logic to update tests and documentation

### CODE REVIEW

File: traffic_ops/traffic_ops_golang/user/user.go [hunk 1]
```diff
@@ -271,6 +271,8 @@ func (user *TOUser) Update(h http.Header) (error, error, int) {
 		return userErr, sysErr, errCode
 	}
 
+	user.RegistrationSent = &tc.TimeNoMod{Time: time.Now()}
```
Comment (mitchell852) [inline traffic_ops/traffic_ops_golang/user/user.go]: confused. updating user (PUT /users/:id) should not update registration_sent value.

Comment (mitchell852) [reply traffic_ops/traffic_ops_golang/user/user.go]: the ONLY way to change that value is by calling `POST users/register`

Comment (mitchell852) [reply traffic_ops/traffic_ops_golang/user/user.go]: oh wait, nm. i guess i'm looking at outdated code :)

Comment (mitchell852) [reply traffic_ops/traffic_ops_golang/user/user.go]: can u change the title of this PR? it confused me.

Comment (rimashah25) [reply traffic_ops/traffic_ops_golang/user/user.go]: this is old code.

**Review by mitchell852 — APPROVED** [review]
lgtm

**Pull request merged**